### PR TITLE
Server theme support

### PR DESCRIPTION
**Pull request by fmoc**
Support server-side provided theme as described in https://github.com/owncloud/ocis/issues/5513.

Depends on #10626. Should eventually fix https://github.com/owncloud/client/issues/10519.

Related: https://github.com/owncloud/ocis/issues/5754

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14706/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14706/oc10/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14706/oc10/guiReportUpload/serverlog.log

Screenshots:
- [Syncing_a_folder_to_the_server.png](https://cache.owncloud.com/public/owncloud/client/14706/oc10/guiReportUpload/screenshots/Syncing_a_folder_to_the_server.png)
- [share_a_file_with_many_users.png](https://cache.owncloud.com/public/owncloud/client/14706/oc10/guiReportUpload/screenshots/share_a_file_with_many_users.png)

</details>

File: src/gui/servertheme.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+class OWNCLOUDSYNC_EXPORT ServerTheme
+{
+    // merely for the enum
+    Q_GADGET
+
+public:
+    enum class Type {
+        Default,
+        Light = Default,
```
Comment (TheOneRing) [inline src/gui/servertheme.h:17]: Why the aliases?

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: src/gui/accountstate.cpp [hunk 2]
```diff
@@ -116,6 +116,22 @@ AccountState::AccountState(AccountPtr account)
     if (FolderMan::instance()) {
         FolderMan::instance()->socketApi()->registerAccount(account);
     }
+
+    connect(this, &AccountState::stateChanged, this, [this](State state) {
```
[the remark this answers is not in the document]
Comment (fmoc) [reply src/gui/accountstate.cpp:120]: I didn't find this to be doable in any other state, but please feel free to suggest alternatives!

File: src/gui/accountsettings.cpp [hunk 3]
```diff
@@ -109,6 +112,35 @@ AccountSettings::AccountSettings(const AccountStatePtr &accountState, QWidget *p
...
+        connect(iconJob, &SimpleNetworkJob::finishedSignal, this, [this, iconJob]() { _accountState->setServerThemeIcon(iconJob->asIcon()); });
+        iconJob->start();
+    });
+
+    if (!_accountState->serverTheme().isEmpty()) {
+        Q_EMIT _accountState->serverThemeChanged();
+    }
+
+    connect(_accountState.data(), &AccountState::serverThemeIconChanged, this, [this]() {
+        const auto &icon = _accountState->serverThemeIcon();
```
[the remark this answers is not in the document]
Comment (fmoc) [reply src/gui/accountsettings.cpp:136]: Not sure what you mean, please elaborate.

**Pull request closed without merging**